### PR TITLE
Vampire branch

### PR DESCRIPTION
**Pull request by btbernheim**
@bontgoy Tagging for a code review!

### CODE REVIEW

**Comment by btbernheim** [comment]
@bontgoy

File: ruby/vampires.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    		name2 = false
+    	elsif name1 == "Tu Fang"
+    		name2 = false
+    		else name2 = true
+    	end
+  
+    puts "How old are you? What year were you born?"
+	age = gets.to_i
+	birth_year = gets.to_i
+  		if birth_year + age == 2016
```
Comment (Bontgoy) [inline ruby/vampires.rb:21]: Consider checking the birth year agains the Ruby time class `Time.now.year` this makes the code more dynamic and allows you the freedom of not having to rewrite the file with hardcoded values as years change.

File: ruby/vampires.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  			age2 = true
+  			else age2 = false
+  		end
+		
+	puts "Our company cafeteria serves garlic bread. Should we order some for you? (y/n)"
+	garlic = gets.chomp
+    	if garlic == "y"
+    		garlic2 = true
+    		else garlic == "n"
```
Comment (Bontgoy) [inline ruby/vampires.rb:30]: `else` does not take a condition.  `garlic == "n"` is not being ran the way you may think.  `else` will run if all other conditions do not pass.  If you want another condition then consider `elsif`.  In this case you are fine just using `else` without the condition.

File: ruby/vampires.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    			garlic2 = false
+    	end
+	
+	puts "Would you like to enroll in the company’s health insurance? (y/n)"
+  	insurance = gets.chomp
+        if insurance == "y"
+        	insurance2 = true
+        	else insurance == "n"
```
Comment (Bontgoy) [inline ruby/vampires.rb:38]: same as above on `else`

File: ruby/vampires.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	    until allergies.include? "sunlight" or allergies.include? "done"
+	      allergies = gets.chomp
+	    end
+	    if allergies.include? "sunlight"
+	    	allergies2 = true
+			else
+		      allergies2 = false
+	    end
+  
+	if !name2
```
Comment (Bontgoy) [inline ruby/vampires.rb:53]: Lets consider order of operations here. I understand that the instructions may be guiding you to just put in conditions as they are mentioned. However, we could clean this up and make the file much more efficient by analyzing our conditions for optimal calls and runtime.

If the employee is named "Drake Cula" or "Tu Fang" then we know that they are definitely a vampire. If that's the case, is there any need to even run the rest of the code?

Consider nesting your conditions and creating a strong order of operations for optimized logic and runtime.
```
if name == "Drake Cula" || name == "Tu Fang"
   "definitely a vampire"
else
  if age_check
    do something
  elsif garlic_and_insurance_check
    do something
  else
    do something
  end
end
```

**Pull request merged**